bazel: add sample-only option to go_gapic_library

idoqo opened the pull request:
Adds the `sample` and `sample-only` protoc plugin options to the bazel rule and should close #233.

noahdietz commented:
I didn't realize this but we need to add _another_ attribute for a plugin flag to enable the sample gen feature.

`gapic-config` is the name of the plugin option, but we should add an optional attribute to the Bazel rule named `gapic_yaml` to stay consistent with other GAPIC rules. It is another `file_args` item.

Essentially, add the following
```
gapic_yaml = None,
...
if gapic_yaml:
    file_args[gapic_yaml] = "gapic-config"
```

I have also updated the test repo, so pull that and build the `echo_go_gapic_sample` target to test! https://github.com/noahdietz/bazel-microgen-test/commit/adeaa41c06b210f9059e06a59b5aeda25e7e51ec

Also, make sure to pull recent changes to this repo to keep up with master :)

idoqo commented:
I've added the gapic_yaml arg and updated the readme too. Sorry, I'm a bit confused about something though - when I generated the samples using protoc (for the bazel-microgen-test repo), I get a `showcase_echo_sample.go` file in `$OUTPUT_DIR/samples` but with bazel, it's just the `srcjar` files in bazel-bin. That means bazel isn't generating those samples right?

noahdietz commented:
> it's just the srcjar files in bazel-bin

Actually, this jar contains all of the generated code! This is a packaging mechanism all of our languages follow. If you list the contents, you will the generated files `jar -tf bazel-bin/echo_go_gapic.srcjar` (or whatever the ja name is).

noahdietz commented on README.md:
```diff
@@ -147,6 +147,10 @@ following attributes:
     * _Note: This option will eventually be deprecated._
+  * `sample`: a path to a sample config file. There 
+  can exist multiples of this.
+  * `gapic_yaml`: path to a gapic configuration file.
```
Can you add a sub-bullet under this one stating it's use is only for sample generation, please? Something like:

* `gapic_yaml`: path to a gapic configuration file.
  * _Note: This config is only used for sample generation._

noahdietz commented on go.mod:
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/russross/blackfriday v2.0.0+incompatible // indirect
 	github.com/shurcooL/sanitized_anchor_name v1.0.0 // indirect
-	google.golang.org/genproto v0.0.0-20200207204624-4f3edf09f4f6
+	google.golang.org/genproto v0.0.0-20200211111953-2dc5924e3898
```
Can you revert the changes to the `go.mod` and `go.sum`, please? We have a bot that updates the deps.

noahdietz commented:
Also Fixes #235. These should've been in the same issue.

noahdietz commented:
Great job @idoqo ! Thank you so much for the contribution, we really appreciate it.
